tapfreighter/interface.go

```diff
@@ -87,8 +87,16 @@ type CoinLister interface {
 	//
 	// If coin selection cannot be completed, then ErrMatchingAssetsNotFound
 	// should be returned.
+	//
+	// NOTE: The coin lock must be held when calling this function.
 	ListEligibleCoins(context.Context,
 		CommitmentConstraints) ([]*AnchoredCommitment, error)
+
+	// WithCoinLock will acquire the coin lock and execute the passed
+	// function. Callers to any coin related functions should use this to
+	// ensure that the coin lock is held across the entire set of
+	// operations.
+	WithCoinLock(fn func() error) error
```

Comment:
I think this placement makes sense, but then I think the implementation can just reside within `CoinSelect ` (the concrete implementation).